src/Sulu/Component/CustomUrl/Document/Subscriber/InvalidationSubscriber.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\CustomUrl\Document\Subscriber;
+
+use PHPCR\Util\PathHelper;
+use Sulu\Bundle\ContentBundle\Document\BasePageDocument;
+use Sulu\Bundle\HttpCacheBundle\Cache\CacheManager;
+use Sulu\Component\CustomUrl\Document\CustomUrlDocument;
+use Sulu\Component\CustomUrl\Document\RouteDocument;
+use Sulu\Component\CustomUrl\Manager\CustomUrlManagerInterface;
+use Sulu\Component\DocumentManager\Event\PersistEvent;
+use Sulu\Component\DocumentManager\Event\PublishEvent;
+use Sulu\Component\DocumentManager\Event\RemoveEvent;
+use Sulu\Component\DocumentManager\Events;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Invalidate custom-url http-cache.
+ */
+class InvalidationSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var CustomUrlManagerInterface
+     */
+    private $manager;
+
+    /**
+     * @var null|CacheManager
+     */
+    private $cacheManager;
+
+    /**
+     * @param CustomUrlManagerInterface $manager
+     * @param null|CacheManager $cacheManager
+     */
+    public function __construct(CustomUrlManagerInterface $manager, ?CacheManager $cacheManager)
+    {
+        $this->manager = $manager;
+        $this->cacheManager = $cacheManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            Events::PERSIST => 'handlePersist',
+            Events::PUBLISH => ['invalidateDocumentBeforePublishing', 1024],
+            Events::REMOVE => ['invalidateDocumentBeforeRemoving', 1024],
+        ];
+    }
+
+    /**
+     * Invalidate custom-urls for persisted pages.
+     *
+     * @param PersistEvent $event
+     */
+    public function handlePersist(PersistEvent $event)
```

Comment:
Do we need that method? I think now that you correctly invalidate the cache on publish it's not needed anymore (I think it is even executed twice now...)